src/exporters/s5.py

```diff
@@ -0,0 +1,492 @@
+
+        selection_dict = {
+            "year": years,
+            "month": months,
+            "day": "01",  # forecast initialised on 1st each month
+        }
+
+        return selection_dict
+
+    @staticmethod
+    def get_6hrly_leadtimes(max_leadtime: int) -> List[int]:
+        # every 6hours up to the max number of days
+        hrly_6 = np.array([6, 12, 18, 24])
+        all_hrs = np.array(
+            [hrly_6 + ((day - 1) * 24) for day in range(1, max_leadtime + 1)]
+        )
+        # prepend the 0th hour to the list
+        leadtime_times = np.insert(all_hrs.flatten(), 0, 0)
+
+        return leadtime_times
+
+    def get_s5_leadtimes(self, max_leadtime: int) -> Dict:
+        """Get the leadtimes for monthly or hourly data"""
+        if self.granularity == "monthly":
+            assert (
+                max_leadtime <= 6
+            ), f"The maximum leadtime is 6 months.\
+                You asked for: {max_leadtime}"
+            leadtime_key = "leadtime_month"
+            leadtime_times = [m for m in range(1, max_leadtime + 1)]
+
+        elif self.granularity == "hourly":
+            assert (
+                max_leadtime <= 215
+            ), f"The maximum leadtime is 215 days.\
+                You asked for: {max_leadtime}"
+            leadtime_key = "leadtime_hour"
+
+            if self.pressure_level:
+                leadtime_times = [day * 24 for day in range(1, 215 + 1)]
+            else:
+                leadtime_times = list(self.get_6hrly_leadtimes(max_leadtime))
+
+            assert (
+                max(leadtime_times) <= 5160
+            ), f"Max leadtime must be less \
+                than 215 days (5160hrs)"
+        else:
+            assert (
+                False
+            ), f'granularity must be in ["monthly", "hourly"]\
+                Currently: {self.granularity}'
+
+        # convert to strings
+        leadtime_times_str = [str(lt) for lt in leadtime_times]
+        selection_dict = {
+            leadtime_key: leadtime_times_str  # leadtime_key differs for monthly/hourly
+        }
+
+        return selection_dict
+
+    def get_product_type(self, product_type: Optional[str] = None) -> Optional[str]:
+        """By default download the 'monthly_mean' product for monthly data """
+        valid_product_types = self.dataset_reference["product_type"]
+
+        if valid_product_types is None:
+            f"{self.dataset} has no `product_type` key. \
+            Only the monthly datasets have product types!"
+            return None
+
+        # if not provided then download monthly_mean
+        if product_type is None:
+            print("No `product_type` provided. Therefore, downloading `monthly_mean`")
+            product_type = "monthly_mean"
+
+        assert (
+            product_type in valid_product_types
+        ), f"Invalid `product_type`: \
+            {product_type}. Must be one of: {valid_product_types}"
+
+        return product_type
+
+    # @staticmethod
```

Comment:
Can this be deleted?

Reply:
if you're happy for me to remove that functionality from the user?

Reply:
sorry just read below